### PR TITLE
Issue deprecation warnings for safe accesses to extern statics

### PR DESCRIPTION
**Pull request by petrochenkov**
Fixes https://github.com/rust-lang/rust/issues/35112
cc https://github.com/rust-lang/rust/issues/36247

### CODE REVIEW

File: src/librustc_lint/lib.rs [hunk 1]
```diff
@@ -207,6 +207,10 @@ pub fn register_builtins(store: &mut lint::LintStore, sess: Option<&Session>) {
             id: LintId::of(LIFETIME_UNDERSCORE),
             reference: "RFC 1177 <https://github.com/rust-lang/rfcs/pull/1177>",
         },
+        FutureIncompatibleInfo {
+            id: LintId::of(SAFE_EXTERN_STATICS),
+            reference: "issue 35112 <https://github.com/rust-lang/rust/issues/35112>",
```
Comment (nikomatsakis) [inline src/librustc_lint/lib.rs]: the new policy is to make a separate tracking issue, following roughly [this template]; if you could update with an issue like that, it'd be great.

[this template]: https://github.com/rust-lang/rfcs/blob/master/text/1589-rustc-bug-fix-procedure.md#tracking-issue-template

**Comment by petrochenkov** [comment]
@nikomatsakis 
Updated.

**Comment by nikomatsakis** [comment]
Looks great. =)

**Pull request merged**